buildSrc/src/main/groovy/org/opensearch/gradle/plugin/OptionalDependenciesPlugin.groovy

```diff
@@ -79,7 +79,7 @@ class OptionalDependenciesPlugin implements Plugin<Project> {
 
                                 if (foundDep) {
                                     if (foundDep.optional) {
-                                        foundDep.optional.value = 'true'
+                                        foundDep.optional[0].value = 'true'
```

Comment:
Could you please explain this change? thank you

Reply:
Without this change `:server:generatePomFileForNebulaPublication` fails because foundDep.optional is of type NodeList. When I print out the value of foundDep, I get:

```
{http://maven.apache.org/POM/4.0.0}dependency[attributes={}; value=[{http://maven.apache.org/POM/4.0.0}groupId[attributes={}; value=[org.apache.logging.log4j]], {http://maven.apache.org/POM/4.0.0}artifactId[attributes={}; value=[log4j-core]], {http://maven.apache.org/POM/4.0.0}version[attributes={}; value=[2.21.0]], {http://maven.apache.org/POM/4.0.0}scope[attributes={}; value=[compile]], {http://maven.apache.org/POM/4.0.0}exclusions[attributes={}; value=[{http://maven.apache.org/POM/4.0.0}exclusion[attributes={}; value=[{http://maven.apache.org/POM/4.0.0}groupId[attributes={}; value=[*]], {http://maven.apache.org/POM/4.0.0}artifactId[attributes={}; value=[*]]]]]], optional[attributes={}; value=true]]]
```

This is the error:

```
> Task :server:generatePomFileForNebulaPublication FAILED
Caching disabled for task ':server:generatePomFileForNebulaPublication' because:
  Task is untracked because: Gradle doesn't understand the data structures used to configure this task
Task ':server:generatePomFileForNebulaPublication' is not up-to-date because:
  Task is untracked because: Gradle doesn't understand the data structures used to configure this task

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:generatePomFileForNebulaPublication'.
> Could not apply withXml() to generated POM
   > argument type mismatch
```

Optional deps are those marked like [this](https://github.com/opensearch-project/OpenSearch/blob/main/server/build.gradle#L106). (f.e. `api "org.locationtech.jts:jts-core:${versions.jts}", optional`)

Reply:
Interesting, thanks @cwperks !